src/reformat.c

```diff
@@ -126,6 +132,24 @@ avifBool avifPrepareReformatState(const avifImage * image, const avifRGBImage *
         }
     }
 
+    if (avifRGBFormatHasAlpha(rgb->format) && !rgb->ignoreAlpha) {
+        state->toYUVPremultiply = image->alphaPremultiplied && !rgb->alphaPremultiplied;
+        state->toYUVUnpremultiply = !image->alphaPremultiplied && rgb->alphaPremultiplied;
+    } else {
+        state->toYUVPremultiply = AVIF_FALSE;
+        state->toYUVUnpremultiply = AVIF_FALSE;
+    }
+
+    if (image->alphaPlane) {
+        avifBool rgbShouldPremultiply = rgb->alphaPremultiplied || !avifRGBFormatHasAlpha(rgb->format) || rgb->ignoreAlpha;
```

Comment:
IMPORTANT: We need some comments for this condition. You can adapt the following from the commit message:

> if we are converting some image with alpha into a format without alpha, we should do 'premultiply alpha' before discarding alpha plane. This has the same effect of rendering this image on a black background, which makes sense.

I also suggest we file an issue and discuss this policy there. I don't know what is the right thing to do in this case.